### DISCUSSION TITLE
100x memory increase when loading a 11mb JPEG into BitmapImage

### DISCUSSION THREAD

**Discussion by NWoodsman** [dotnet/wpf#9434] [Q&A]
I've added a sample project and the jpeg image (Arc de Triomphe, from Wikipedia). Running the generated .exe in release shows about 32mb -> 320mb when the image is loaded. 

All the code is in MainWindow.xaml.cs.

I believe I am using `BitmapImage` pretty canonically (not doing weird stuff). 

https://github.com/NWoodsman/ImageMemoryDemo

**Comment by yll690** [comment]
The resolution of your picture is too high . The memory usage by image can be calculated by `width*height*bytesPerPixel*2`. You can set BitmapImage.DecodePixelWidth to a smaller value like 1000 to save memory and decoding time.
